### PR TITLE
Pr code quality

### LINKED ISSUE #102: Testcase fix
Failing testcases (that have been commented out):

JsonSerializableModuleBookTest: toModelType_typicalTasksFile_success()

StorageManagerTest: moduleBookReadSave()

JsonModuleBookStorageTest: readAndSaveModuleBook_allInOrder_success()

### LINKED ISSUE #106: Bug report
After recur a task that startTime do not have a time field, the time field will somehow occur

### LINKED ISSUE #109: Bug report
StartTime and Deadline both are checked, which should not be that case.
If now is 18:00 1/3/2021. StartTime is 15:00 1/2/2021 and deadline is 00:00 2/3/2021. After refresh, startTime will be later than deadline.

### PR DESCRIPTION
**Pull request by AiwassPrime**
Issues:

> Issue #103 AddCommand: Line 58
> Issue #103 EditCommand: Line 90

Bugs fixed:

> Issue #102 All test case fixes
> Issue #106 fixes. Task: Line 143
> Issue #109 fixes. Task: Line 205

### CODE REVIEW

**Review by yhtMinceraft1010X — COMMENTED** [review]
Nice! LGTM although I recommend adding OptionalFieldTest unit tests.

**Comment by AiwassPrime** [comment]
> Nice! LGTM although I recommend adding OptionalFieldTest unit tests.

It is implemented.

File: src/main/java/seedu/module/logic/commands/AddCommand.java [hunk 1]
```diff
@@ -55,12 +55,17 @@ public AddCommand(Task task) {
...
         requireNonNull(model);
-        if (model.hasTask(toAdd) && !toAdd.isRecurring()) {
+        /*
+        if (!toAdd.isRecurring() && model.hasTask(toAdd)) {
             throw new CommandException(MESSAGE_DUPLICATE_TASK);
         }
         if (toAdd.isRecurring() && model.hasRecurringTask(toAdd)) {
             throw new CommandException(RecurCommand.MESSAGE_DUPLICATE_RECURRENCE);
         }
+        */
```
Comment (umergta) [inline src/main/java/seedu/module/logic/commands/AddCommand.java:65]: maybe can remove this commented code out

File: src/main/java/seedu/module/logic/commands/DeleteTagCommand.java [hunk 2]
```diff
@@ -72,29 +65,7 @@ public CommandResult execute(Model model) throws CommandException {
...
-        } else if (!taskToTag.isRecurring() && taskToTag.isDeadline()) {
-            editedTask = new Task(name, deadline, module, description, workload, newDoneStatus, newTags);
-
-        } else if (taskToTag.isRecurring() && !taskToTag.isDeadline()) {
-            editedTask = new Task(name, startTime, deadline, module, description, workload, newDoneStatus, recurrence,
-                    newTags);
-
-        } else {
-            editedTask = new Task(name, deadline, module, description, workload, newDoneStatus, recurrence, newTags);
-        }
```
Comment (umergta) [inline src/main/java/seedu/module/logic/commands/DeleteTagCommand.java:97]: nice haha

**Review by umergta — COMMENTED** [review]
LGTM! Moving forward, think there are 2 things to do for me and yuanzhe..
1. implement a way to remove recurrence
2. implement a way to remove starttime
(Since both are optional fields)

3. Add more tests (not so urgent as can be done in 1.4)

**Review by QY-H00 — COMMENTED** [review]
LGTM Overall. Nice bro!

**Pull request merged**